src/timeline/RoomlistModel.cpp

```diff
@@ -1105,3 +1123,100 @@ FilteredRoomlistModel::previousRoom()
         }
     }
 }
+
+#ifdef NHEKO_DBUS_SYS
+RoomListDBusInterface::RoomListDBusInterface(RoomlistModel *parent)
+  : QObject{parent}
+  , m_parent{parent}
+{
+    connect(ChatPage::instance(), &ChatPage::newRoom, this, &RoomListDBusInterface::prepareModel);
+    connect(ChatPage::instance(), &ChatPage::leftRoom, this, &RoomListDBusInterface::prepareModel);
+    prepareModel();
+}
+
+QVector<nheko::dbus::RoomInfoItem>
+RoomListDBusInterface::getRooms(const QDBusMessage &message)
+{
+    auto reply = message.createReply();
+
+    m_modelAccess.lock();
+    nhlog::ui()->debug("Sending {} rooms over D-Bus...", m_model->size());
+    reply << QVariant::fromValue(*m_model);
+    QDBusConnection::sessionBus().send(reply);
+    nhlog::ui()->debug("Rooms successfully sent to D-Bus.");
+    m_modelAccess.unlock();
+
+    return {};
+}
+
+void
+RoomListDBusInterface::activateRoom(const QString &roomid) const
+{
+    bringWindowToTop();
+    m_parent->setCurrentRoom(roomid);
+}
+
+void
+RoomListDBusInterface::joinRoom(const QString &roomid) const
```

Comment:
Is this actually the roomid or the alias?

Reply:
Currently it is the alias